packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -502,6 +506,27 @@ export async function createRouteManifest(
 	}
 
 	const redirectRoutes = createRedirectRoutes(params, routeMap, logger);
+	if (dev && config.base === '/') {
+		// Add a redirect for `//` to `/`. We only do this in dev because different platforms
+		// vary a lot in how or whether they handle this sort of redirect, so we leave it to the adapter
+		redirectRoutes.push({
+			type: 'redirect',
+			isIndex: false,
+			route: '//',
+			pattern: /^\/+$/,
+			segments: [],
+			params: [],
+			component: '/',
+			generate: () => '//',
+			pathname: '//',
+			prerender: false,
+			redirect: '/',
+			redirectRoute: routeMap.get('/'),
+			fallbackRoutes: [],
+			distURL: [],
+			origin: 'project',
+		});
```

Comment:
I would advise against creating full routes like this because many of those data might not be mandatory, and some of them need to be built using the proper functions. E.g. `getPattern` or `getRouteGenerator`.

Plus, we need to take into consideration things like `base`, `trailing slash` and more. Follow this pattern:

https://github.com/withastro/astro/blob/901c21f4f09bf61dfbb5ab04db2d7d90120383cb/packages/astro/src/core/routing/manifest/create.ts#L237-L261

It requires more code, but it's more reliable

Reply:
I tried that at first but it didn't seem to be possible for this. Unless I'm missing something, the path isn't one that can be represented with segments. It's for a specific path, regardless of trailing slash or base rules. The pattern needs to match two or more slashes, whether or not trailing slash is enabled, and doesn't apply if base is set. Of the fields I included, it looks like `pathname` is the only one that's optional.